tests/agent_test.go

```diff
@@ -0,0 +1,235 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	"github.com/gkampitakis/go-snaps/match"
+	"github.com/gkampitakis/go-snaps/snaps"
+	"github.com/stretchr/testify/assert"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+func assertLogsContainsString(t *testing.T, logs io.Reader, expected string) {
+	t.Helper()
+
+	// FIXME: Is there a better strategy to read the logs?
+	buf := new(bytes.Buffer)
+
+	_, err := buf.ReadFrom(logs)
+	assert.NoError(t, err)
+
+	text := buf.String()
```

Comment:
I think you can use a `strings.Builder` instead of `bytes.Buffer`, which can be more efficient and performant.

1. https://golangnote.com/golang/golang-stringsbuilder-vs-bytesbuffer
2. https://medium.com/@felipedutratine/string-concatenation-in-golang-since-1-10-bytes-buffer-vs-strings-builder-2b3081848c45